### ISSUE TITLE
FNX2-13050 ⁃ [Bug] Changing TalkBack Setting While Browsing Doesn't Change Toolbar Scroll Flags

### ISSUE THREAD

**Issue by ekager** [mozilla-mobile/fenix#10026]
## Steps to reproduce
1. Go to any website without TalkBack on
1. Turn TalkBack on via a shortcut without leaving browsing

### Expected behavior
Toolbar readjusts scroll flags to new a11y setting

### Actual behavior
Toolbar does not readjust. 

I think we just have to add an Accessibility Service Listener in the browser that changes the settings and flags responsively to fix this. We used to have this working in Focus IIRC

### Device information
* Android device: ?
* Fenix version: ?

**Comment by abodea** [comment]
Hi, @ekager note that I re-checked this issue with the latest **Nightly 7/27** with **Google Pixel 4 XL (Android 11).**
Please note the following: Following the steps from the description after enabling the TalkBack nothing major happened I was able to normally scroll and the Toolbar disappeared when I scrolled down and appeared when I scrolled up.
Is this the intended behavior? I don't know exactly how I can check or see the Toolbar Scroll Flags.
[Video](https://drive.google.com/file/d/10GFO-r909E-lXTqwDzDZZAJX-PSQrzYy/view).

**Comment by abodea** [comment]
**Verified** as **fixed** on the latest **Nightly 7/30** with **Google Pixel 4 XL (Android 11).**
I checked the following scenarios:
 - When the talkback is ON before launching Fenix, the toolbar doesn't disappear when scrolling, but if deactivating without leaving the app, the toolbar is scrolled again.
- When the talkback is OFF and reaching any website the toolbar disappears, after activating the talkback toolbar doesn't disappear anymore.
